### PR TITLE
[WIP] Expose minimal API for flutter_tools consumption

### PR DESCRIPTION
**Pull request by jonahwilliams**
This allows flutter tools to inject its own serving logic for certain assets.

### CODE REVIEW

File: webdev/lib/src/serve/webdev_server.dart [hunk 1]
```diff
@@ -22,12 +22,14 @@ class ServerOptions {
   final int port;
   final String target;
   final int daemonPort;
+  final Handler handler;
```
Comment (jakemac53) [inline webdev/lib/src/serve/webdev_server.dart]: nit: `customHandler` or something a bit more descriptive

File: webdev/lib/src/serve/webdev_server.dart [hunk 2]
```diff
@@ -22,12 +22,14 @@ class ServerOptions {
...
 
   ServerOptions(
     this.configuration,
     this.port,
     this.target,
     this.daemonPort,
+    this.handler,
```
Comment (jakemac53) [inline webdev/lib/src/serve/webdev_server.dart]: nit: make nullable args optional (named)

**Comment by jonahwilliams** [comment]
Re-using this to discuss some ideas with the API structure

File: webdev/lib/webdev.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import 'src/serve/debugger/devtools.dart';
+import 'src/serve/webdev_server.dart';
+
+export 'src/command/configuration.dart' show Configuration, ReloadConfiguration;
+export 'src/daemon_client.dart' show daemonPort;
+export 'src/serve/chrome.dart' show Chrome, ChromeError;
+export 'src/serve/debugger/app_debug_services.dart' show AppDebugServices;
+export 'src/serve/debugger/devtools.dart' show DevTools;
+
+Future<WebDevHandler> connectToWebdev(
```
Comment (jonahwilliams) [inline webdev/lib/webdev.dart]: This is the first stab at a basic API - I tried to expose as little as possible - really only the configuration needed to initialize the server and the interfaces we use to get the vm service or debug. See https://github.com/flutter/flutter/pull/34252/files#diff-01a7653ade6cc1a5182fdebf929ad64bR291 for example usage

File: webdev/lib/webdev.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    assetPort,
+    optionalHandler,
+  );
+  var webDevServer =
+      await WebDevServer.start(serverOptions, buildResults, devTools);
+  var chrome = await Chrome.start(
+      <String>['http://${webDevServer.host}:${webDevServer.port}/'],
+      port: configuration.chromeDebugPort);
+  var devHandler = webDevServer.devHandler;
+  var connection = await devHandler.connectedApps.first;
```
Comment (jakemac53) [inline webdev/lib/webdev.dart]: Do we want to support multiple apps at once? I think from a webdev perspective it makes sense to support that but maybe not from a flutter perspective.

Essentially that would mean the WebDevHandler would have a stream of application handlers or something like that.

**Pull request closed without merging**